portfolio/src/main/webapp/script.js

```diff
@@ -31,23 +31,93 @@ function addRandomGreeting() {
+  let numberOfComments = dropdownNumber.options[dropdownNumber.selectedIndex].text;
 
-  // Response received in the form: {"commentsArray": [{name: "", comment:"", rating:}, {}, {}]}.
-  fetch("/comments").then(response => response.json()).then((commentList) => {
+  if (numberOfComments == "all") {
+    numberOfComments = "0";
+  }
+
+  // Hardcoded URL with requsted number of comments.
+  let URL = "/comments?number=" + numberOfComments;
+
+  // Response received in the form: [{name: "", comment:"", rating:}, {}, {}].
+  fetch(URL).then(response => response.json()).then((commentList) => {
+    console.log("Recieve request with comments array")
     const commentsListElement = document.getElementById('comment-field-table');
     commentsListElement.innerHTML = '';
 
     for (i in commentList) {
-      commentsListElement.appendChild(createListElement('Name: ' + commentList[i].name));
-      commentsListElement.appendChild(createListElement('Comment text: ' + commentList[i].text));
-      commentsListElement.appendChild(createListElement('Rating: ' + commentList[i].rating));
-      commentsListElement.appendChild(createListElement('\n'));
+      commentsListElement.appendChild(createListElement(
+        commentList[i].name, commentList[i].text, commentList[i].rating)
+      );
     }
   });
 }
 
-function createListElement(text) {
-  const liElement = document.createElement('li');
-  liElement.innerText = text;
-  return liElement;
+function createListElement(name, text, rating) {
+  let newCommentBox = document.createElement('div');
+  newCommentBox.className = "comment-box";
+
+  let topSection = document.createElement('div');
+  topSection.id = "top-section";
+
+  let newName = document.createElement('p');
+  newName.id = "name-section";
+  newName.innerHTML = name;
+
+  topSection.appendChild(newName);
+
+  let ratingSection = document.createElement('div');
+  ratingSection.id = "rating-section";
+  
+  let newRating = document.createElement('p');
+  newRating.innerHTML = "Rating→" + rating.toString();
+
+  ratingSection.appendChild(newRating);
+  topSection.appendChild(ratingSection);
+
+  let textSection = document.createElement('div');
+  textSection.id = "text-section";
+
+  let newText = document.createElement('p');
+  newText.innerHTML = text;
+
+  textSection.appendChild(newText);
+
+  newCommentBox.appendChild(topSection);
+  newCommentBox.appendChild(textSection);
+
+  newCommentBox.id = "shown-comment";
+
+  return newCommentBox;
+}
+
+function toggleCommentSection() {
+  let displayButton = document.getElementsByClassName('display-comments-buttons')[0];
+  let commentSection = document.getElementsByClassName('comment-section')[0];
+
+  if (commentSection.style.width == "450px") {
+    commentSection.style.width = "0%";
+    displayButton.innerHTML = "Show comment section";
+
+    let comments = document.getElementById("shown-comment");
+    while (comments != null) {
```

Comment:
I am not 100% clear on the logic. When toggling the comments off, this seems to delete all comments, but I do not see where they are added back in in the else (when toggling them back on).

Is there a need to delete the comments? It might be simpler to just set height of the comments section to 0 (or set the display property to `none' (see https://www.w3schools.com/cssref/pr_class_display.asp) when you hide the comments and restore it to its previous value when you show them again.

Reply:
I can explain what I meant here! Without that code when person close and then reopen comment section, previously shown comments still will be visible. I didn't want that (I don't remember why) so I decided to delete comments once person close that section. If person want to see comments again they should request them again. 

Without that code comments still won't be seen when person close section.
(Sorry for sloppy explanation I can show examples of what I mean)